clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -108,6 +116,19 @@ public boolean isMute() {
         return muted;
     }
 
+    public boolean isInMutableState() {
+        //some requests (like disconnect) may not require memory allocation.
+        //do we do not want to just mute channels where otherwise the next request received
+        //could be immediately executed regardless of memory.
+        //a channel would mute itself if it cannot allocate memory.
+        //also, we cannot mute channels where any memory required has already been allocated.
+        //(for requests that dont require memory receive.memoryAllocated() would also be true
+        if (receive == null || receive.memoryAllocated())
+            return false;
+        //also cannot mute if underlying transport is unmutable right now
+        return transportLayer.isInMutableState();
```

Comment:
It may be better to call `this.ready()` rather than `transportLayer.ready()` authenticators don't use the memory pool and channels don't need to be muted during authentication.